mlrun/api/crud/model_monitoring/_model_endpoint_stores/sql_model_endpoint_store.py

```diff
@@ -0,0 +1,551 @@
+        endpoint_list = mlrun.api.schemas.model_endpoints.ModelEndpointList(
+            endpoints=[]
+        )
+
+        # Validate that the model endpoints table exists
+        if not engine.has_table(self.table_name):
+            logger.warn(
+                "Table not found, return an empty ModelEndpointList",
+                table=self.table_name,
+            )
+            return endpoint_list
+
+        with engine.connect():
+            # Generate the sqlalchemy.schema.Table object that represents the model endpoints table
+            metadata = db.MetaData()
+            model_endpoints_table = db.Table(
+                self.table_name, metadata, autoload=True, autoload_with=engine
+            )
+
+            # Get the model endpoints records using sqlalchemy ORM
+            session = sessionmaker(bind=engine)()
+
+            columns = model_endpoints_table.columns.keys()
+            query = session.query(model_endpoints_table).filter_by(project=self.project)
+
+            # Apply filters
+            if model:
+                query = self._filter_values(
+                    query=query,
+                    model_endpoints_table=model_endpoints_table,
+                    key_filter=model_monitoring_constants.EventFieldType.MODEL,
+                    filtered_values=[model],
+                )
+            if function:
+                query = self._filter_values(
+                    query=query,
+                    model_endpoints_table=model_endpoints_table,
+                    key_filter=model_monitoring_constants.EventFieldType.FUNCTION,
+                    filtered_values=[function],
+                )
+            if uids:
+                query = self._filter_values(
+                    query=query,
+                    model_endpoints_table=model_endpoints_table,
+                    key_filter=model_monitoring_constants.EventFieldType.ENDPOINT_ID,
+                    filtered_values=uids,
+                    combined=False,
+                )
+            if top_level:
+                node_ep = str(mlrun.model_monitoring.EndpointType.NODE_EP.value)
+                router_ep = str(mlrun.model_monitoring.EndpointType.ROUTER.value)
+                endpoint_types = [node_ep, router_ep]
+                query = self._filter_values(
+                    query=query,
+                    model_endpoints_table=model_endpoints_table,
+                    key_filter=model_monitoring_constants.EventFieldType.ENDPOINT_TYPE,
+                    filtered_values=endpoint_types,
+                    combined=False,
+                )
+            # Labels from type list won't be supported from 1.4.0
+            # TODO: Remove in 1.4.0
+            if labels and isinstance(labels, typing.List):
+                logger.warning(
+                    "Labels should be from type dictionary, not list,"
+                    "Will be deprecated in 1.4.0.",
+                    labels=labels,
+                )
```

Comment:
We are moving to a deprecation model where we raise a `FutureWarning` - see https://jira.iguazeng.com/browse/ML-3104 for some more details. In general you should use the `warnings` package for deprecation warnings.